compiler/luci/lang/include/luci/IR/AttrDilation.h

```diff
@@ -27,15 +28,25 @@ class Dilation final
 public:
   Dilation() : _w(1), _h(1) {}
 
-  int32_t w() const { return _w; }
-  void w(int32_t w) { _w = w; }
-
-  int32_t h() const { return _h; }
-  void h(int32_t h) { _h = h; }
+  uint32_t w() const { return _w; }
+  void w(uint32_t w) { _w = w; }
+  void w(int32_t w)
+  {
+    assert(w >= 0);
+    _w = static_cast<uint32_t>(w);
+  }
```

Comment:
(Optional) Why don't you hide cassert include in .cpp?

Reply:
@parjong Just curiosity, is there any reason of this? Is that for good looking?

Reply:
> Just curiosity, is there any reason of this? Is that for good looking?

The goal is to minimize impact on namespace (or symbol table), which eventually minimizes the coupling between components.

The current implementation allows clients (= the code that includes this header) to use assert without including 'cassert' header.

This may introduce implicit coupling between two components, which makes it a bit difficult to update this header later. 

For example, let us assume that we want to remove assert here. Suddenly, build breaks from other components because they uses assert without ``#include <cassert>`` statement.